File: src/main/java/org/funivan/lologa/algo/ga/genome/ScoreCalculation.java
package org.funivan.lologa.algo.ga.genome;

import org.funivan.lologa.algo.ga.genome.value.ValueInterface;

import java.util.HashMap;

public class ScoreCalculation {

    private final GenomeInterface genome;

    public ScoreCalculation(GenomeInterface genome) {
        this.genome = genome;
    }

    public Double calculate(final HashMap<String, Double> metric) {
        Double score = 0.0;
        final HashMap<String, Double> data = this.genome.data();
        for (final String type : data.keySet()) {
            if (metric.containsKey(type)) {
                score = score + metric.get(type) * data.get(type);
            }
        }
        return score;
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/ScoreCalculation.kt
package org.funivan.lologa.algo.ga.genome

import java.util.*

class ScoreCalculation(private val genome: HashMap<String, Double>) {

    fun calculate(metric: HashMap<String, Double>): Double? {
        var score: Double? = 0.0
        val data = this.genome
        for (type in data.keys) {
            if (metric.containsKey(type)) {
                score = score!! + metric[type] * data[type]
            }
        }
        return score
    }
}
